modules/nextflow/src/main/groovy/nextflow/scm/RepositoryProvider.groovy

```diff
@@ -242,7 +242,7 @@ abstract class RepositoryProvider {
     }
 
     @Memoized
-    protected <T> List<T> invokeAndResponseWithPaging(String request, Closure<T> parse) {
+    protected List invokeAndResponseWithPaging(String request, Closure parse) {
```

Comment:
This method seems to be the issue. You can either remove `@Memoized` or remove the generics, and the relevant tests stop failing. 🤷

Reply:
Good finding, but it's a pity we need to remove the generic type